keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,525 @@
+                    @Override
+                   public Mono<? extends RestResponse<byte[]>> apply(RestResponse<SecretBackup> base64URLRestResponse) {
+                        return Mono.just(new SimpleRestResponse<byte[]>(base64URLRestResponse.request(),
+                            base64URLRestResponse.statusCode(), base64URLRestResponse.headers(), base64URLRestResponse.body().value()));
+                    }
+                });
+    }
+
+    /**
+     * Restores a backed up secret, and all its versions, to a vault.
+     * This operation requires the secrets/restore permission.
+     *
+     * @param backup The backup blob associated with the secret.
+     * @throws NullPointerException if {@code SecretBackup} is {@code null}.
+     * @return A {@link Mono} containing a {@link RestResponse} whose {@link RestResponse#body()} contains the restored {@link Secret}.
+     * @throws com.azure.common.http.rest.RestException when the {@code backup} is corrupted.
+     */
+    public Mono<RestResponse<Secret>> restoreSecretAsync(byte[] backup) {
+        Objects.requireNonNull(backup, "The Secret backup parameter cannot be null.");
+        SecretRestoreRequestParameters parameters = new SecretRestoreRequestParameters()
+                                                .secretBackup(backup);
+        return service.restoreSecret(vaultEndPoint, API_VERSION, ACCEPT_LANGUAGE, parameters);
+    }
+
+    /**
+     * List {@link Secret secrets} in the key vault.
+     * The list Secrets operation is applicable to the entire vault. However, only the base secret identifier and its attributes are provided in the response.
+     * Individual secret versions are not listed in the response. This operation requires the secrets/list permission.
+     *
+     * @return A {@link Flux} containing {@link SecretAttributes} of all the secrets in the vault.
+     */
+    public Flux<SecretAttributes> listSecretsAsync() {
+        return getPagedSecrets(service.getSecrets(vaultEndPoint, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE));
+    }
+
+    /**
+     * Lists {@link DeletedSecret secrets} of the key vault.
+     * The get deleted secrets operation returns the secrets that have been deleted for a vault enabled for soft-delete.
+     * This operation requires the secrets/list permission.
+     *
+     * @return A {@link Flux} containing all of the {@link DeletedSecret deleted secrets} in the vault.
+     */
+    public Flux<DeletedSecret> listDeletedSecretsAsync() {
+        return getPagedDeletedSecrets(service.getDeletedSecrets(vaultEndPoint, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE));
+    }
+
+    /**
+     * List all versions of the specified secret. The individual secret version response in the list is represented by {@link SecretAttributes}.
+     * The full secret identifier and attributes are provided in the response. No values are returned for the secrets.
+     * This operation requires the secrets/list permission.
+     *
```

Comment:
In these kind of JavaDocs, it would be useful to indicate to the user how they take a `SecretAttributes` instance and get a full `Secret` out of it - if that is the common thing that they will need to do...

Reply:
Added to javadocs, that Secret Attributes represents the secret response from Azure KV as the response only contains base id and attributesof  the secret properties/details and not its value.

What exactly do you mean by getting a full Secret out of SecretAttrbiutes?
Casting SecretAttributes to Secret?

Reply:
I haven't reminded myself of the API, but presumably the user will need to do some call in the SecretClient to getSecret(SecretAttributes), for example?

Reply:
Oh okay,
I've updated javadocs, with code samples on how they can get list of Secrets with values after getting list of SecretAttributes.